aioredis/commands/streams.py

```diff
@@ -0,0 +1,222 @@
+
+    return [_list_to_dict(l) for l in lists]
+
+
+class StreamCommandsMixin:
+    """Stream commands mixin
+
+    Streams are under development in Redis and
+    not currently released.
+    """
+
+    def xadd(self, stream, fields, message_id=b'*', max_len=None,
+             exact_len=False):
+        """ Add a message to the specified stream
+        """
+        args = []
+        if max_len is not None:
+            if exact_len:
+                args.extend((b'MAXLEN', max_len))
+            else:
+                args.extend((b'MAXLEN', b'~', max_len))
+
+        args.append(message_id)
+
+        for k, v in fields.items():
+            args.extend([k, v])
+        return self.execute(b'XADD', stream, *args)
+
+    def xrange(self, stream, start='-', stop='+', count=None):
+        """Retrieve stream data"""
+        if count is not None:
+            extra = ['COUNT', count]
+        else:
+            extra = []
+        fut = self.execute(b'XRANGE', stream, start, stop, *extra)
+        return wait_convert(fut, parse_messages)
+
+    def xrevrange(self, stream, start='+', stop='-', count=None):
+        """Retrieve stream data"""
+        if count is not None:
+            extra = ['COUNT', count]
+        else:
+            extra = []
+        fut = self.execute(b'XREVRANGE', stream, start, stop, *extra)
+        return wait_convert(fut, parse_messages)
+
+    def xread(self, streams, timeout=0, count=None, latest_ids=None):
+        """Perform a blocking read on the given stream"""
+        args = self._xread(streams, timeout, count, latest_ids)
+        fut = self.execute(b'XREAD', *args)
+        return wait_convert(fut, parse_messages_by_stream)
+
+    def xread_group(self, group_name, consumer_name, streams, timeout=0,
+                    count=None, latest_ids=None):
+        args = self._xread(streams, timeout, count, latest_ids)
+        fut = self.execute(
+            b'XREADGROUP', b'GROUP', group_name, consumer_name, *args
+        )
+        return wait_convert(fut, parse_messages_by_stream)
+
+    def xgroup_create(self, stream, group_name, latest_id='$'):
+        fut = self.execute(b'XGROUP', b'CREATE', stream, group_name, latest_id)
+        return wait_ok(fut)
+
+    def xgroup_setid(self, stream, group_name, latest_id='$'):
+        fut = self.execute(b'XGROUP', b'SETID', stream, group_name, latest_id)
+        return wait_ok(fut)
+
+    def xgroup_delgroup(self, stream, group_name):
+        fut = self.execute(b'XGROUP', b'DELGROUP', stream, group_name)
+        return wait_ok(fut)
+
+    def xgroup_delconsumer(self, stream, consumer_name):
+        fut = self.execute(b'XGROUP', b'DELCONSUMER', stream, consumer_name)
+        return wait_ok(fut)
+
+    def xpending(self, stream, group_name, start=None, stop=None, count=None,
+                 consumer=None):
+        # Returns: total pel messages, min id, max id, count
+        ssc = [start, stop, count]
+        if any(ssc) and not all(ssc):
```

Comment:
This check may trigger false-positive exceptions in case when `start=0`.
Please add test for this case.

Reply:
Good catch! Will add test and fix